crates/l2/sequencer/l1_committer.rs

```diff
@@ -104,58 +90,111 @@ impl Committer {
     }
+}
 
-    pub async fn run(&mut self) {
-        loop {
-            if let Err(err) = self.main_logic().await {
-                error!("L1 Committer Error: {}", err);
-            }
+#[derive(Clone)]
+pub enum InMessage {
+    Watch,
+}
 
-            sleep_random(self.commit_time_ms).await;
-        }
+#[allow(dead_code)]
+#[derive(Clone, PartialEq)]
+pub enum OutMessage {
+    Done,
+    Error,
+}
+
+pub struct L1Committer;
+
+impl L1Committer {
+    pub async fn spawn(
+        store: Store,
+        rollup_store: StoreRollup,
+        execution_cache: Arc<ExecutionCache>,
+        cfg: SequencerConfig,
+    ) -> Result<(), SequencerError> {
+        let state = CommitterState::new(
+            &cfg.l1_committer,
+            &cfg.eth,
+            store.clone(),
+            rollup_store.clone(),
+            execution_cache.clone(),
+        )?;
+        let mut l1_committer = L1Committer::start(state);
+        let _ = l1_committer.cast(InMessage::Watch).await;
+        Ok(())
     }
+}
 
-    async fn main_logic(&mut self) -> Result<(), CommitterError> {
-        // Get the batch to commit
-        let last_committed_batch_number = self
-            .eth_client
-            .get_last_committed_batch(self.on_chain_proposer_address)
-            .await?;
-        let batch_to_commit = last_committed_batch_number + 1;
-
-        // Get the last committed block_number
-        let last_committed_block_number = self
-            .rollup_store
-            .get_block_numbers_by_batch(last_committed_batch_number)
-            .await?
-            .and_then(|blocks| blocks.last().copied())
-            .ok_or_else(|| CommitterError::InternalError("Invalid rollup_storage state".into()))?;
-
-        let first_block_to_commit = last_committed_block_number + 1;
-
-        // Try to prepare batch
-        let (
-            blobs_bundle,
-            new_state_root,
-            withdrawal_hashes,
-            deposit_logs_hash,
-            last_block_of_batch,
-        ) = self
-            .prepare_batch_from_block(last_committed_block_number)
-            .await?;
+impl GenServer for L1Committer {
+    type InMsg = InMessage;
+    type OutMsg = OutMessage;
+    type State = CommitterState;
+
+    type Error = CommitterError;
+
+    fn new() -> Self {
+        Self {}
+    }
 
-        if last_committed_block_number == last_block_of_batch {
-            debug!("No new blocks to commit, skipping");
-            return Ok(());
+    async fn handle_call(
+        &mut self,
+        _message: Self::InMsg,
+        _tx: &Sender<GenServerInMsg<Self>>,
+        _state: &mut Self::State,
+    ) -> CallResponse<Self::OutMsg> {
+        CallResponse::Reply(OutMessage::Done)
+    }
+
+    async fn handle_cast(
+        &mut self,
+        _message: Self::InMsg,
+        tx: &Sender<GenServerInMsg<Self>>,
+        state: &mut Self::State,
+    ) -> CastResponse {
+        // Right now we only have the watch message, so we ignore the message
```

Comment:
I like this... actually I should change `L1Watcher` since it also receives only one kind of message